### ISSUE TITLE
9da4edb broke my layout

### ISSUE THREAD

**Issue by RocusHalbasch** [Benoker/DockingFrames#66]
In our code we implemented a custom WorkingArea that implements MultipleCDockable. In it's init sets the PlacholderStrategy to a null strategy with the following code:

```
getStation().setPlaceholderStrategy(new PlaceholderStrategy() {
	@Override
	public void uninstall(DockStation station) {}

	@Override
	public void removeListener(PlaceholderStrategyListener listener) {}

	@Override
	public boolean isValidPlaceholder(Path placeholder) {
		return true;
	}

	@Override
	public void install(DockStation station) {}

	@Override
	public Path getPlaceholderFor(Dockable dockable) {
		return null;
	}

	@Override
	public void addListener(PlaceholderStrategyListener listener) {}
});
```

This has worked for years to prevent placeholders from being stored. However with the 9da4edb commit this no longer worked. Now if you open two tabs you get a StackDockStation as you always have but when you close one of them SplitDockStation.replace is called to replace the StackDockStation with a DefaultCommonDockable which in turn calls leaf.setDockable(next, null, true, station) this results in placeholders being stored despite our null strategy above due to the code changes from the commit. And from that point forward our application no longer behaves correctly if only one tab is open in our working area and a Dockable is added instead of creating a StackDockStation and placing both tabs in it it now splits the working area and adds the new dockable in the new area.

**Comment by Benoker** [comment]
I did not think of such a side effect. I'll try to reproduce your issue and find a bugfix for both issues.

**Comment by Benoker** [comment]
There is a really easy solution: the method `isValidPlaceholer` of your custom `PlaceholderStrategy` should just return `false`.

Why? Because the `PlaceholderMap` that is read from the removed station is filtered using your `PlaceholderStrategy`. And 9da4edb reads the placeholders from the map **after** is was filtered.

**Comment by RocusHalbasch** [comment]
Thanks. That is simple enough. Weird that it worked that way for years.

**Comment by Benoker** [comment]
You (mis)used the bug I fixed in 9da4edb :-)

Let me know if you run in any more issues.
